cs-rin-ru-enhanced-mod.user.js

```diff
@@ -487,10 +495,8 @@ function setupTopicPreview() {
         const topic = $(e)[0];
         let tid;
         $(topic).off("mouseover").on("mouseover", () => {
-            showPreview = true;
```

Comment:
Actually I think this variable is needed because without it in some specific conditions (it's hard to reproduce) when you hover mouse over topic and then move it away, the preview pops up anyway.

Reply:
I don't know if I'm experiencing the same situation as you, but I seem to get it with or without this variable anyway. I think it has something to do with the logic of loading the preview. Currently, the code waits for the user to hover over the topic for a certain amount of time, and then it will get the page, which also takes some time. 

I think the problem occurs when the user hovers over the topic over the specified amount of time and then moving the mouse away, which would cause the code to load the page because the timer has expired. 

Basically, if I set the timer to 1 second, and loading the page takes 2 seconds, if I hover over the link for 1 second, the timer will expire and the `setTimeout` code will run, and once the `GM_xmlhttpRequest` is sent, it can't be cancelled. I did some testing, and it seems that hovering over tags seems to lead to unintended behaviour (like loading the wrong page, sending multiple requests and loading the tags of other topics the user hovers over).

Reply:
> I think the problem occurs when the user hovers over the topic over the specified amount of time and then moving the mouse away, which would cause the code to load the page because the timer has expired.

That's what this variable was for, so after the page preview is loaded we check again if user still keeps mouse cursor over topic.
Don't know about tags, they should be a part of topic title so they should work in the same way.

Reply:
Anyway current version seems to work fine.

Reply:
> That's what this variable was for, so after the page preview is loaded we check again if user still keeps mouse cursor over topic. Don't know about tags, they should be a part of topic title so they should work in the same way.

Yeah, I just modified the code a little so it should work as expected.